File: level 068/classwork/classwork.js
// 8 Kyu JavaScript

// 1
// https://www.codewars.com/kata/57a2013acf1fa5bfc4000921/train/javascript

function findAverage(array) {
    if (array.length == 0) {
      return 0
    }
    else{
      const sum = array.reduce((partialSum, a) => partialSum + a, 0);
      return sum / array.length;
    }
}

// 2
// https://www.codewars.com/kata/53dc23c68a0c93699800041d/train/javascript

function smash (words) {
    return words.join(" ");
};

// 3
// https://www.codewars.com/kata/57eadb7ecd143f4c9c0000a3/train/javascript

function abbrevName(name){
    let text = name.split(" ");
    let firstName = text[0];
    let lastName = text[1];
    let initials = [];
    initials.push(firstName[0].toUpperCase());
    initials.push(lastName[0].toUpperCase());
    let result = initials.join(".");
    return result;
};

// 4
// https://www.codewars.com/kata/5556282156230d0e5e000089/train/javascript

function DNAtoRNA(dna) {
    let rna = "";
    for (let i of dna){
      if (i != "T"){
        rna = rna + i;
      }
      else{
        rna = rna + "U";
      }
    }
    return rna;
}

// 5
// https://www.codewars.com/kata/54ff3102c1bad923760001f3/train/javascript

function getCount(str) {
  const vowels = "aeiou";
  let count = 0;
  for (let i of str){
    if (vowels.includes(i)){
      count++;
    }
  }
  return count;
};

File: level 034/classwork/classwork.py
# https://www.codewars.com/kata/5556282156230d0e5e000089/train/python

def dna_to_rna(dna):
    rna = ""
    for i in dna:
        if i != "T":
            rna += i
        else:
            rna += "U"
    return rna

# https://www.codewars.com/kata/57cebe1dc6fdc20c57000ac9/train/python

def find_short(s):
    list1 = s.split(" ")


    word_len = len(list1[0])
    for i in list1:
        if len(i) < word_len:
            word_len = len(i)
    
    return word_len